### PR TITLE
Add GIFs to our request documentation

### LINKED ISSUE #108: Investigate the possibility of adding GIFs to YARD documentation
It's easier to see and understand documented features visually, rather than explain LSP requests in text.

Explore the possibility of adding GIFs to our requests documentation. E.g.:
- Are they displayed properly when generated from YARD?
- Where should GIFs be stored?

### PR DESCRIPTION
**Pull request by vinistock**
### Motivation

Closes #108

Add GIFs to our request documentation to better show which features are supported.

### Implementation

By commit
1. Include the `misc` folder in YARD generation (assets just get copied over)
2. Exclude `misc` from the gemspec, so that users don't have to download the GIFs
3. Adapt `check_docs` to verify that, instead of written examples, the documentation includes a GIF
4. Change all request documentation to include GIFs
5. Add the `misc` folder with all of the request GIFs
6. Add a list of supported features in `RubyLsp::Requests` that we can point to from the README
7. Enforce that all requests are present in the `RubyLsp::Requests` list in `check_docs`

### Manual Tests

1. Switch to this branch
2. `bundle exec rake yard`
3. In a browser, use CMD + SHIFT + O and open the index file under `....../ruby-lsp/docs/index.html`
4. Navigate in the documentation for specific requests and check the GIFs

Alternatively, you can also start yard in server mode, but don't forget to pass it all of the options we have listed in the `Rakefile`.

### CODE REVIEW

File: lib/ruby_lsp/requests/code_actions.rb [hunk 1]
```diff
@@ -3,17 +3,11 @@
...
     # request informs the editor of RuboCop quick fixes that can be applied. These are accesible by hovering over a
     # specific diagnostic.
-    #
-    # # Example
-    #
-    # ```ruby
-    # def say_hello
-    # puts "Hello" # --> code action: quick fix indentation
-    # end
-    # ```
```
[the remark this answers is not in the document]
Comment (vinistock) [reply lib/ruby_lsp/requests/code_actions.rb:16]: That's a good point. Do you have suggestions on how to write the examples in an accessible format? Are the old ones good enough?

[the remark this answers is not in the document]
Comment (vinistock) [reply lib/ruby_lsp/requests/code_actions.rb:16]: That's totally okay. And thank you for bringing this up!

**Review by RyanBrushett — COMMENTED** [review]
Thanks @egiurleo! Excellent points.

Should we keep the written code-examples for now until we do a proper investigation into this? Also maybe worth mentioning in favour of keeping both written examples and gifs: Gifs are big/slow to load. Anyone on spotty internet/mobile internet/etc. might have trouble with those.

**Review by wildmaples — APPROVED** [review]
GIFs look great!!! Is it also worth documenting the keys you need to hit for stuff like selection ranges?

**Pull request merged**